service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/TopicProducerActor.scala

```diff
@@ -52,79 +88,108 @@ private[lagom] class TaggedOffsetProducerActor[Message](
     extends Actor
     with ActorLogging {
 
-  /** Switch used to terminate the on-going Kafka publishing stream when this actor fails.*/
+  /** Switch used to terminate the on-going stream when this actor is stopped.*/
   private var shutdown: Option[KillSwitch] = None
 
   override def postStop(): Unit = {
     shutdown.foreach(_.shutdown())
   }
 
   override def receive = {
-    case EnsureActive(tag) =>
-      val daoFuture = offsetStore.prepare(s"topicProducer-$topicId", tag)
-
-      // null or empty strings become None, otherwise Some[String]
-      def strToOpt(str: String) =
-        Option(str).filter(_.trim.nonEmpty)
-
-      val serviceLookupFuture: Future[Option[String]] =
-        kafkaConfig.serviceName match {
-          case Some(name) =>
-            locateService(name)
-              .map { uris =>
-                strToOpt(UriUtils.hostAndPorts(uris))
-              }
-
-          case None =>
-            Future.successful(strToOpt(kafkaConfig.brokers))
-        }
-
-      serviceLookupFuture.zip(daoFuture).pipeTo(self)
-
-      context.become(initializing(tag))
-  }
-
-  def generalHandler: Receive = {
-    case Status.Failure(e) => throw e
-    case EnsureActive(_)   =>
+    case EnsureActive(tagName) =>
+      self ! Start
+      context.become(initializing(tagName))
   }
 
-  private def initializing(tag: String): Receive = generalHandler.orElse {
+  private def initializing(tagName: String): Receive = {
```

Comment:
The bulk of the change happens on `receive`, `initializing` and (the now gone) `start`.

Replicating, once more, the code in `ReadSideImpl` the default `receive` handles `EnsureActive` and sets everything up. Since `TopicProducer` doesn't need any `globalPrepare` the `receive` becomes trivial. Then, `Start` can be sent to self.

`initializing` (which I'll rename to started) handles `Start` to create a `RestartSource.withBackoff` which locates brokers, loads the offset and uses those futures to create a query by tag (`eventStreamFactory`) and connect it to user code (`eventsPublisherFlow`).

The remainder of changes in this file are mostly shuffling code around and trying to make it look like `ReadSideImpl` for future refactors.